src/composables/queries/usePoolsQuery.ts

```diff
@@ -92,7 +102,6 @@ export default function usePoolsQuery(
           ])
         );
         await injectTokens(tokens);
-        await forChange(dynamicDataLoading, false);
```

Comment:
Unrelated with current PR: 
I find this `forChange` function too complex. Maybe we could also replace the other usages?

Reply:
hmm I find it quite useful and pretty self-explanatory. Do you mean the internal logic? If you read that line as a kind of sentence it makes sense:

Await for `dynamicDataLoading` to change to `false`

Reply:
Yeah, I meant the internal logic. Now I read it again and made more sense. Thanks.